### PR TITLE
init sequence.md chapter

### PR DESCRIPTION
**Pull request by AndrewDi**
init sequence.md chapter

### CODE REVIEW

File: session1/chapter7/sequence.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+```SQL
+SELECT SETVAL(sequence_name,100)；
+```
+
+本章将会通过一些案例介绍 TiDB Sequence 的使用方法：
+
+1. 并发的应用需要获取单调递增的流水号
+
+在使用分布式数据库的场景里，通常应用也是分布式架构，这样多个应用节点之间如何获取唯一且递增的序列号就成为一个难题。在分布式数据库没有 Sequence 的时候，应用基本通过`雪花算法`、`数据库主键自增`等方法实现，业界也有一些较为成熟的方案，比如  [Leaf - 美团点评分布式 ID](https://tech.meituan.com/2017/04/21/mt-leaf.html)、[百度的 uid-generator](https://github.com/baidu/uid-generator)等，上述方案中为了解决单调递增且不重复的问题引入一个新的系统、模块，极大的提高的应用系统的复杂度。这里通过简单新建一个`Sequence`看看 TiDB 如何解决上述问题。
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
在使用分布式数据库的场景里，通常应用也是分布式架构，这样多个应用节点之间如何获取唯一且递增的序列号就成为一个难题。在分布式数据库没有 Sequence 的时候，应用基本通过`雪花算法`、`数据库主键自增`等方法实现，业界也有一些较为成熟的方案，比如 [Leaf - 美团点评分布式 ID](https://tech.meituan.com/2017/04/21/mt-leaf.html)、[百度的 uid-generator](https://github.com/baidu/uid-generator) 等，上述方案中为了解决单调递增且不重复的问题引入一个新的系统、模块，极大的提高的应用系统的复杂度。这里通过简单新建一个 `Sequence` 看看 TiDB  如何解决上述问题。
```

File: session1/chapter7/sequence.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+1.1. 首先新建一个 Sequence
+
+```SQL
+CREATE SEQUENCE seq_for_unique START WITH 1 INCREMENT BY 1 CACHE 1000 NOCYCLE;
+```
+
+1.2. 针对应用连接至单个 TiDB 和多个 TiDB，取到的 Sequence 值有些不一样
+
+1） 如果两个应用节点同时连接只同一个`TiDB`节点，两个节点间取到的则为连续递增的值
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
1） 如果两个应用节点同时连接只同一个 TiDB 节点，两个节点间取到的则为连续递增的值
```

File: session1/chapter7/sequence.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+| NEXT VALUE FOR seq_for_unique |
++-------------------------------+
+|                          1001 |
++-------------------------------+
+1 row in set (0.00 sec)
+```
+
+2. 在一张表里面需要有多个自增字段
+
+MySQL 语法中每张表仅能新建一个`auto_increment`字段，且该字段必须为主键，在应用设计的时候，主键通常有业务意义的字段表示，例如用户名、主机名等，但通过 Sequence 和生成列，我们可以实现这个需求。
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
MySQL 语法中每张表仅能新建一个 `auto_increment` 字段，且该字段必须定义在主键或是唯一索引列上，在应用设计的时候，主键通常有业务意义的字段表示，例如用户名、主机名等，但通过 Sequence 和生成列，我们可以实现这个需求。
```

File: session1/chapter7/sequence.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+2.1. 首先新建如下两个 Sequence
+
+```SQL
+CREATE SEQUENCE seq_for_autoid START WITH 1 INCREMENT BY 2 CACHE 1000 NOCYCLE;
+CREATE SEQUENCE seq_for_logid START WITH 100 INCREMENT BY 1 CACHE 1000 NOCYCLE;
+```
+
+2.2. 在新建表的时候通过`default nextval(seq_name)`设置列的默认值
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
2.2. 在新建表的时候通过 `default nextval(seq_name)` 设置列的默认值
```

File: session1/chapter7/sequence.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+2.3. 接下来我们插入几个用户信息进行测试：
+
+```SQL
+INSERT INTO user (userid) VALUES ('usera');
+INSERT INTO user (userid) VALUES ('userb');
+INSERT INTO user (userid) VALUES ('userc');
+```
+
+2.4. 查询`user`表，可以发现`autoid`和`logid`字段的值按照不同的步长进行自增，且主键仍然在列`userid`上：
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
2.4. 查询 `user` 表，可以发现 `autoid` 和 `logid` 字段的值按照不同的步长进行自增，且主键仍然在列 `userid` 上：
```

File: session1/chapter7/sequence.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+| usera  |      1 |   100 |
+| userb  |      3 |   101 |
+| userc  |      5 |   102 |
++--------+--------+-------+
+3 rows in set (0.01 sec)
+```
+
+3. 更新数据表中其中一列值为连续自增的值
+
+假设我们有一张数据表，表中有 20 万的数据，我们需要更新其中一列的值为连续自增且唯一的整数，如果没有 Sequence，我们只能通过应用一条条读写记录，并用`update`更新值，并且还需要按照固定批次提交，但现在我们有了 Sequence，一切都会变的特别简单。
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
假设我们有一张数据表，表中有 20 万行数据，我们需要更新其中一列的值为连续自增且唯一的整数，如果没有 Sequence，我们只能通过应用一条条读写记录，并用 `update` 更新值，并且还需要按照固定批次提交，但现在我们有了 Sequence，一切都会变的特别简单。
```

File: session1/chapter7/sequence.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+3.4. 更新为连续的值
+
+```SQL
+tidb[test]> update t set a=nextval(test);
+Query OK, 10000 rows affected (0.20 sec)
+Rows matched: 10000  Changed: 10000  Warnings: 0
+```
+
+3.5. 查询结果集,可以看到字段`a`的值已经连续自增且唯一
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
3.5. 查询结果集,可以看到字段 `a` 的值已经连续自增且唯一
```

File: session1/chapter7/sequence.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+|    17 | user17    |
+|    18 | user18    |
+|    19 | user19    |
+|    20 | user20    |
+...
+```
+
+## 注意事项
+
+在分布式架构的数据库中实现完成连续递增的序列是比较有难度的，而`Sequence`把`严格递增`和`性能`两方面交给了使用者，在新建`Sequence`的时候，可以通过组合`Order/No Order`（目前不支持）和`Cache/No Cache`来选择新建高性能的`Sequence`,亦或是性能较差但递增较为严格的`Sequence`。
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
在分布式架构的数据库中实现完成连续递增的序列是比较有难度的，而`Sequence`把`严格递增`和`性能`两方面交给了使用者，在新建 `Sequence` 的时候，可以通过组合 `Order/No Order`（目前尚未实现）和 `Cache/No Cache` 来选择新建高性能的 `Sequence`，亦或是性能较差但递增较为严格的 `Sequence`。
```

File: session1/chapter7/sequence.md [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+
+在使用分布式数据库的场景里，通常应用也是分布式架构，这样多个应用节点之间如何获取唯一且递增的序列号就成为一个难题。在分布式数据库没有 Sequence 的时候，应用基本通过`雪花算法`、`数据库主键自增`等方法实现，业界也有一些较为成熟的方案，比如 [Leaf - 美团点评分布式 ID](https://tech.meituan.com/2017/04/21/mt-leaf.html)、[百度的 uid-generator](https://github.com/baidu/uid-generator) 等，上述方案中为了解决单调递增且不重复的问题引入一个新的系统、模块，极大的提高的应用系统的复杂度。这里通过简单新建一个 `Sequence` 看看 TiDB  如何解决上述问题。
+
+1.1. 首先新建一个 Sequence
+
+```SQL
+CREATE SEQUENCE seq_for_unique START WITH 1 INCREMENT BY 1 CACHE 1000 NOCYCLE;
+```
+
+1.2. 针对应用连接至单个 TiDB 和多个 TiDB，取到的 Sequence 值有些不一样
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
1.2. 从不同的 TiDB 节点获取到的 Sequence 值顺序有所不同
```

File: session1/chapter7/sequence.md [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+
+1） 如果两个应用节点同时连接只同一个 TiDB 节点，两个节点间取到的则为连续递增的值
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
1） 如果两个应用节点同时连接至同一个 TiDB 节点，两个节点间取到的则为连续递增的值
```

File: session1/chapter7/sequence.md [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+
+```SQL
+SELECT SETVAL(sequence_name,100)；
+```
+
+本章将会通过一些案例介绍 TiDB Sequence 的使用方法：
+
+1. 并发的应用需要获取单调递增的流水号
+
+在使用分布式数据库的场景里，通常应用也是分布式架构，这样多个应用节点之间如何获取唯一且递增的序列号就成为一个难题。在分布式数据库没有 Sequence 的时候，应用基本通过`雪花算法`、`数据库主键自增`等方法实现，业界也有一些较为成熟的方案，比如 [Leaf - 美团点评分布式 ID](https://tech.meituan.com/2017/04/21/mt-leaf.html)、[百度的 uid-generator](https://github.com/baidu/uid-generator) 等，上述方案中为了解决单调递增且不重复的问题引入一个新的系统、模块，极大的提高的应用系统的复杂度。这里通过简单新建一个 `Sequence` 看看 TiDB  如何解决上述问题。
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
在使用分布式数据库的场景里，通常应用也是分布式架构，这样多个应用节点之间如何获取唯一且递增的序列号就成为一个难题。在分布式数据库没有 Sequence 的时候，应用基本通过`雪花算法`、`数据库主键自增`等方法实现，业界也有一些较为成熟的方案，比如 [Leaf - 美团点评分布式 ID](https://tech.meituan.com/2017/04/21/mt-leaf.html)、[百度的 uid-generator](https://github.com/baidu/uid-generator) 等，上述方案中为了解决单调递增且不重复的问题引入一个新的系统、模块，极大的提高的应用系统的复杂度。这里通过简单新建一个 `Sequence` 看看 TiDB 如何解决上述问题。
```

File: session1/chapter7/sequence.md [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+| NEXT VALUE FOR seq_for_unique |
++-------------------------------+
+|                          1001 |
++-------------------------------+
+1 row in set (0.00 sec)
+```
+
+2. 在一张表里面需要有多个自增字段
+
+MySQL 语法中每张表仅能新建一个 `auto_increment` 字段，且该字段必须定义在主键或是唯一索引列上，在应用设计的时候，主键通常有业务意义的字段表示，例如用户名、主机名等，但通过 Sequence 和生成列，我们可以实现这个需求。
```
Comment (bb7133) [inline session1/chapter7/sequence.md]:
```suggestion
MySQL 语法中每张表仅能新建一个 `auto_increment` 字段，且该字段必须定义在主键或是唯一索引列上。在应用设计的时候，主键通常有业务意义的字段表示，例如用户名、主机名等，但通过 Sequence 和生成列，我们可以实现多自增字段需求。
```

**Comment by bb7133** [comment]
Rest LGTM

**Comment by bb7133** [comment]
LGTM

**Review by winkyao — APPROVED** [review]
LGTM

**Pull request merged**